desk/faux/ted/fetch-messages.hoon

```diff
@@ -30,17 +42,25 @@
     ==
   ;<  ~  bind:m  (send-request request)
   ;<  =client-response:iris  bind:m  take-client-response
-  ?>  ?=(%finished -.client-response)
+  ?.  ?=(%finished -.client-response)
+    ~&  >>  not-finished+client-response  (pure:m !>(%not-finished))
+  ?~  full-file.client-response  ~&  >>  rate-limited+client-response  (pure:m !>(%rate-limited))
   =/  raw-body
-    ?~  full-file.client-response  !!
     q.data.u.full-file.client-response
   =/  json-body  (need (de-json:html raw-body))
-  =/  messages  (messages-from-json:faux-discord json-body)
-  ?~  messages
+  =/  messages  (mule |.((messages-from-json:faux-discord json-body)))
+  =/  [ok=? =tang res=(list message:faux-discord)]
+    ?-  -.messages
+      %|  :-  |  :_  ~  (welp p.messages leaf+"retry" ~)
+      %&  ?:  =(~ p.messages)
+        :-  &  :_  ~  `tang`[leaf+"no messages" ~]
+      :-  &  :_  p.messages  (flop `tang`[leaf+"got messages" ~])
```

Comment:
i don't know what the hell i was thinking here

Reply:
it's a bit much